src/utils/matches.ts

```diff
@@ -1,7 +1,26 @@
+import path from "path"
 import micromatch from "micromatch"
 
 export default function matches(paths: string[], patterns: string[]): string[] {
+	// https://github.com/micromatch/micromatch/#options implies that `basename` only applies to patterns without slashes
+	// However, that doesn't seem to be the case in practice; this means that files.matches does not match when
+	//     path=some_directory/File.tsx
+	//     pattern=some_directory/**/*
+	// We ignore patterns with subdirectories for now, pending resolution of https://github.com/micromatch/picomatch/issues/89
+	let hasBasename = false
+	let hasNested = false
+	for (const pattern of patterns) {
+		const isBasename = path.basename(pattern) === pattern
+		hasBasename ||= isBasename
+		hasNested ||= !isBasename
+	}
```

Comment:
Should this loop just be a `.find()` so that it short-circuits?

Reply:
I think I'd need to do two `find`s, to check for both nested and basename. And if we're generally expecting only a couple (in most cases one) pattern, this could be slightly better. wdyt?

Reply:
Probably not a huge deal, doubt you're gonna have that many patterns.

Reply:
Makes sense, switched to `find` for clarity.

Reply:
@jamiebuilds hi hi! Could you help me figure out how to get this bugfix published? 🙏